tests/unit/installer/test_workflows.py

```diff
@@ -13,10 +13,15 @@
 from databricks.labs.ucx.installer.workflows import DeployedWorkflows, WorkflowsDeployment
 
 
+class ResourceDoesNotExistIter:
+    def __iter__(self):
+        raise ResourceDoesNotExist("logs")
+
+
 def test_deployed_workflows_handles_log_folder_does_not_exists(mock_installation):
     ws = create_autospec(WorkspaceClient)
     ws.jobs.list_runs.return_value = [BaseRun(run_id=456)]
-    ws.workspace.list.side_effect = ResourceDoesNotExist("logs")
+    ws.workspace.list.return_value = ResourceDoesNotExistIter()
```

Comment:
This change shouldn't be necessary?
> **`side_effect`**<br/>
> This can either be a function to be called when the mock is called, an iterable **or an exception (class or instance) to be raised**.

Ref: https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.side_effect

Reply:
I think it does

Reply:
Ahh. This is quite subtle:

```suggestion
    # Raise the error when the result is iterated over, NOT when the method is called.
    ws.workspace.list.return_value = ResourceDoesNotExistIter()
```